npm/vite-dev-server/src/resolveConfig.ts

```diff
@@ -31,10 +32,48 @@ export const createConfig = async ({ options, viteConfig: viteOverrides = {} }:
     Falling back to Vite\'s defaults.`)
   }
 
+  debug(`Resolving to Vue, React, and Mount Utils packages here:`,
+    join(binaryRoot, 'npm/vue'),
+    join(binaryRoot, 'npm/react'),
+    join(binaryRoot, 'npm/mount-utils'))
+
   const config = {
     root,
     base: `/${options.config.namespace}/src/`,
     configFile,
+    server: {
+      fs: {
+        // https://vitejs.dev/config/#server-fs-allow
+        //
+        // By default, Vite only allows dependency scanning within the user's own project directories.
+        // Cypress will load npm/* projects as external modules from the binary's location.
+        // Because the Cypress binary is not located within the user's node_modules,
+        // Vite needs to be explicitly allowed to traverse the cypressBinaryRoot directory.
+        //
+        // If `resolve.alias` did not map to a cypressBinaryRoot outside of the user's own project,
+        // then this line would not be necessary.
+        allow: [
+          searchForWorkspaceRoot(process.cwd()),
+          binaryRoot,
+        ],
+      },
+    },
+    resolve: {
```

Comment:
Will there be any problem with someone implementing a third party dev server - will they need to do a similar alias, or is this only needed for dev-servers bundled with the binary?

Reply:
Yes, if someone implements a third-party dev server they'd need to do this as well, or they'd need to use the `@cypress/` bundled packages. 

I guess that's another potential benefit of exporting the actual rollup dist files from the CLI rather than relying on this approach.

Reply:
Hm, so are we going with the proposed approach here? Or do we want to explore alternatives? Or a combination? I think first class third party support is pretty important - it's going to be much more likely an ecosystem emerges if we have seamlessly third party integration.